File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const str = input[0].split("");
const bomb = input[1].split("");
const m = bomb.length;

const stack = [];

for (let i = 0; i < str.length; i++) {
  stack.push(str[i]);

  if (str[i] === bomb[m - 1]) {
    if (stack.length >= m && stack.slice(-m).join("") === bomb.join("")) {
      for (let j = 0; j < m; j++) stack.pop();
    }
  }
}

if (stack.length === 0) console.log("FRULA");
else console.log(stack.join(""));

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
string = list(input())
bomb = list(input())

b_len = len(bomb)
b_start = bomb[0]
r_bomb = bomb[::-1]

stack = []
for i in range(len(string)-1, -1, -1):
    stack.append(string[i])

    if len(stack) >= b_len and string[i] == b_start:
        if stack[-b_len:] == r_bomb:
            for _ in range(b_len):
                stack.pop()

if len(stack) == 0:
    print('FRULA')
else:
    print(''.join(stack[::-1]))
